### PR TITLE
Adding support for 220250 Rev4A

### PR DESCRIPTION
**Pull request by mzieg**
- adds get/set_laser_power_attenuator (new feature on Rev4A boards)
- automatically reads configured laser TEC setpoint from EEPROM and applies to the board on connection
- tweaks to battery read-out
- minor cleanup

### CODE REVIEW

**Comment by mzieg** [comment]
still need to add support for turning the laser TEC on / off, graph laser thermistor in Factory view, etc

File: wasatch/FeatureIdentificationDevice.py [hunk 1]
```diff
@@ -279,28 +279,42 @@ def _post_connect(self): # -> SpectrometerResponse
...
-
-            log.debug("enabling detector TEC")
-            self.detector_tec_setpoint_has_been_set = True
-            self.set_tec_enable(True)
+        if self.settings.is_xs():
+            
+            # XS only supports a TEC on the laser, not the detector
+            if self.settings.eeprom.sig_laser_tec:
+                log.debug("initializing laser TEC on XS")
+                self.set_laser_temperature_setpoint_raw(self.settings.eeprom.startup_temp_degC)
```
Comment (mzieg) [inline wasatch/FeatureIdentificationDevice.py:290]: This is the key change needed for this release

**Comment by mzieg** [comment]
@samiebee43, any issues?  This was used for 4.0.15.

**Review by samiebee43 — APPROVED** [review]
Likewise, this approved by virtue of being in a release.

This should be merged with [#E265](https://github.com/WasatchPhotonics/ENLIGHTEN/pull/265) or [#E279](https://github.com/WasatchPhotonics/ENLIGHTEN/pull/279)

**Pull request merged**